source/extensions/http/header_validators/envoy_default/http2_header_validator.cc

```diff
@@ -116,13 +115,20 @@ Http2HeaderValidator::validateRequestHeaderMap(::Envoy::Http::RequestHeaderMap&
             UhvResponseCodeDetail::get().InvalidMethod};
   }
 
-  auto is_connect_method = header_map.method() == header_values_.MethodValues.Connect;
-  auto is_options_method = header_map.method() == header_values_.MethodValues.Options;
+  // The CONNECT method with the ":protocol" header is called the extended CONNECT and covered in
+  // https://datatracker.ietf.org/doc/html/rfc8441#section-4
+  // For the purposes of header validation the extended CONNECT is treated as generic (non CONNECT)
+  // HTTP/2 requests.
+  auto is_standard_connect_request = header_map.method() == header_values_.MethodValues.Connect &&
+                                     header_map.getProtocolValue().empty();
+  auto is_extended_connect_request = header_map.method() == header_values_.MethodValues.Connect &&
+                                     !header_map.getProtocolValue().empty();
+  auto is_options_request = header_map.method() == header_values_.MethodValues.Options;
   bool path_is_empty = path.empty();
   bool path_is_asterisk = path == "*";
   bool path_is_absolute = !path_is_empty && path.at(0) == '/';
 
-  if (!is_connect_method && (header_map.getSchemeValue().empty() || path_is_empty)) {
+  if (!is_standard_connect_request && (header_map.getSchemeValue().empty() || path_is_empty)) {
```

Comment:
Comment for later, but it'd be nice if UHV either had utilities for these things, or had a state map returned with headers so we can guarantee that later bits in the filter chain come up with the same reuslt.

I believe we've previously had issues where some bits of the code would check one header but not both for example.

Reply:
I have moved these checks into HeaderUtility class. Perhaps to make it less error prone they should be part of the header map class itself.